File: generate-voiceover.mjs
#!/usr/bin/env node
/**
 * Generate voiceover for Elevate for Humanity video
 * Uses the repository's TTS service
 */

import { generateAndSaveAudio } from './server/tts-service.ts';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));

// Compelling script about Elevate for Humanity
const SCRIPT = `Welcome to Elevate for Humanity.

We're more than a training provider. We're a community-based workforce philanthropy, transforming lives through education and opportunity.

From Marion County to communities across Indiana, we connect people to 100% funded career training. No tuition. No debt. Just real pathways to meaningful careers.

Whether you're starting fresh, making a change, or getting a second chance, we're here to elevate your future.

Because when you rise, we all rise.

Elevate for Humanity. Your partner in career transformation.`;

async function main() {
  console.log('🎙️  Generating voiceover for Elevate for Humanity\n');
  console.log('Script:');
  console.log('─'.repeat(70));
  console.log(SCRIPT);
  console.log('─'.repeat(70));
  console.log('\n');

  try {
    const outputPath = path.join(__dirname, 'public', 'videos', 'voiceover.mp3');
    
    console.log('Generating audio with espeak-ng...\n');
    
    await generateAndSaveAudio(SCRIPT, outputPath, {
      voice: 'nova', // Female voice
      speed: 1.0
    });
    
    console.log('\n✅ Voiceover generated successfully!');
    console.log(`\nFile saved to: ${outputPath}`);
    console.log('\nNext steps:');
    console.log('1. Combine voiceover with video using FFmpeg');
    console.log('2. Add background music (optional)');
    console.log('3. Replace the video on the website');
    
  } catch (error) {
    console.error('\n❌ Error generating voiceover:', error.message);
    console.error('\nMake sure espeak-ng and ffmpeg are installed:');
    console.error('  sudo apt-get install espeak-ng ffmpeg');
    process.exit(1);
  }
}

main();

File: generate-voiceover.py
#!/usr/bin/env python3
"""
Generate natural-sounding voiceover using Google Text-to-Speech
Requires: pip install gtts
"""

from gtts import gTTS
import os

# Read the script
script_path = 'public/videos/homepage-voiceover-natural.txt'
output_path = 'public/videos/voiceover-new.mp3'

with open(script_path, 'r') as f:
    text = f.read()

print("🎤 Generating voiceover...")
print(f"📝 Script length: {len(text)} characters")

# Create TTS object with natural settings
# Using 'en' for US English, slow=False for natural pace
tts = gTTS(text=text, lang='en', slow=False, tld='com')

# Save the audio file
tts.save(output_path)

print(f"✅ Voiceover generated: {output_path}")
print(f"📊 File size: {os.path.getsize(output_path) / 1024:.1f} KB")
print("\n🎯 Next steps:")
print("1. Listen to the file: public/videos/voiceover-new.mp3")
print("2. If you like it, replace the old one:")
print("   mv public/videos/voiceover-new.mp3 public/videos/voiceover.mp3")
print("3. Re-enable in components/WelcomeAudio.tsx")
